File: winery-frontend/src/types/vinograd.ts
export interface Vinograd {
  idv: number
  naziv: string
  datosn: string
  povrsina: number
  nadmorskavis: number
  tipzemljista: string
  navodnjavanje: string
  brojParcela: number
}

export interface VinogradDetail {
  idv: number
  naziv: string
  datosn: string
  povrsina: number
  nadmorskavis: number
  tipzemljista: string
  navodnjavanje: string
  parcele: Parcela[]
}

export interface Parcela {
  idp: number
  brojcokota: number
  povrsina: number
  nazivparcele: string
  vinogradIdv: number
  sortagrozdjaIdsrt?: number
  sortaNaziv?: string
}

export interface CreateVinogradRequest {
  naziv: string
  datosn: string
  povrsina: number
  nadmorskavis: number
  tipzemljista: string
  navodnjavanje: string
  parcele?: CreateParcelaRequest[]
}

export interface CreateParcelaRequest {
  brojcokota: number
  povrsina: number
  nazivparcele: string
  sortagrozdjaIdsrt?: number
}

export interface UpdateVinogradRequest {
  naziv: string
  datosn: string
  povrsina: number
  nadmorskavis: number
  tipzemljista: string
  navodnjavanje: string
}

export interface UpdateParcelaRequest {
  brojcokota: number
  povrsina: number
  nazivparcele: string
  sortagrozdjaIdsrt?: number
}

File: WineryAPI/Models/Vinograd.cs
using System;
using System.Collections.Generic;

namespace WineryAPI.Models;

public partial class Vinograd
{
    public int Idv { get; set; }

    public string Naziv { get; set; } = null!;

    public DateOnly Datosn { get; set; }

    public decimal Povrsina { get; set; }

    public int Nadmorskavis { get; set; }

    public string Tipzemljista { get; set; } = null!;

    public char Navodnjavanje { get; set; }

    public virtual ICollection<Parcela> Parcelas { get; set; } = new List<Parcela>();
}
